lms/djangoapps/progress/management/commands/recalculate_progress_for_users.py

```diff
@@ -0,0 +1,103 @@
+"""
+One-time data migration script -- shoulen't need to run it again
+"""
+import logging
+from optparse import make_option
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.db.models import Q
+
+from progress.models import StudentProgress, CourseModuleCompletion
+from progress.signals import is_valid_progress_module
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Recalculate progress entries for the specified course(s) and/or user(s)
+    """
+
+    def handle(self, *args, **options):
+        option_list = BaseCommand.option_list + (  # pylint: disable=unused-variable
+            make_option(
+                "-c",
+                "--course_ids",
+                dest="course_ids",
+                help="List of courses for which to Recalculate progress",
+                metavar="first/course/id,second/course/id"
+            ),
+            make_option(
+                "-u",
+                "--user_ids",
+                dest="user_ids",
+                help="List of users for which to Recalculate progress",
+                metavar="1234,2468,3579"
+            ),
+        )
+
+        course_ids = options.get('course_ids')
+        user_ids = options.get('user_ids')
+
+        status_summary = {'skipped': 0, 'updated': 0}
+        total_users_processed = 0
+        detached_categories = getattr(settings, 'PROGRESS_DETACHED_CATEGORIES', [])
+        cat_list = [Q(content_id__contains=item.strip()) for item in detached_categories]
+        cat_list = reduce(lambda a, b: a | b, cat_list)
+
+        # Get the list of courses from the system
+        courses = modulestore().get_courses()
+
+        # If one or more courses were specified by the caller, just use those ones...
+        if course_ids is not None:
+            filtered_courses = []
+            for course in courses:
+                if unicode(course.id) in course_ids.split(','):
+                    filtered_courses.append(course)
+            courses = filtered_courses
+
+        for course in courses:
+            users = CourseEnrollment.objects.users_enrolled_in(course.id)
+            # If one or more users were specified by the caller, just use those ones...
+            if user_ids is not None:
+                filtered_users = []
+                for user in users:
+                    if str(user.id) in user_ids.split(','):
```

Comment:
Nit: If there are a large number of enrollments or a large number of users specified on the command line, you should move `user_ids.split(',')` outside of the loop, and store it as a `frozenset` for performance.